modules/monitoring-modifying-retention-time-and-size-for-prometheus-metrics-data.adoc

```diff
@@ -0,0 +1,135 @@
+If this size limit is reached, the oldest data is deleted first until the disk space used by the data is again below the maximum.
+
+Note the following behaviors of these data retention settings:
+
+* The retention settings apply only to data block directories. 
+The settings are not used for data located in the `wal` and `chunks_head` directories.
+* The retention settings are used only when Prometheus writes a new data block to a data block directory, which occurs every two hours and after the write-ahead log (WAL) has recorded at least three hours of data.
+* If you do not explicitly define values for either `retention` or `retentionSize`, retention time defaults to 15 days, and retention size is not set.
+* If you define values for both `retention` and `retentionSize`, both values apply.
+If any data blocks exceed either the defined retention time or the defined size limit, then these data blocks are purged.
+* If you define a value for `retentionSize` and do not define `retention`, only the `retentionSize` value applies.
+* If you do not define a value for `retentionSize` and only define a value for `retention`, only the `retention` value applies
+
+.Prerequisites
+
+* *If you are configuring core {product-title} monitoring components*:
+** You have access to the cluster as a user with the `cluster-admin` role.
+** You have created the `cluster-monitoring-config` `ConfigMap` object.
+* *If you are configuring components that monitor user-defined projects*:
+** You have access to the cluster as a user with the `cluster-admin` role, or as a user with the `user-workload-monitoring-config-edit` role in the `openshift-user-workload-monitoring` project.
+** You have created the `user-workload-monitoring-config` `ConfigMap` object.
+* You have installed the OpenShift CLI (`oc`).
+
+.Procedure
+
+. Edit the `ConfigMap` object:
+** *To modify the retention time and size for the Prometheus instance that monitors core {product-title} projects*:
+.. Edit the `cluster-monitoring-config` `ConfigMap` object in the `openshift-monitoring` project:
++
+[source,terminal]
+----
+$ oc -n openshift-monitoring edit configmap cluster-monitoring-config
+----
+
+.. Add the retention time and size configuration under `data/config.yaml`:
++
+[source,yaml]
+----
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: cluster-monitoring-config
+  namespace: openshift-monitoring
+data:
+  config.yaml: |
+    prometheusK8s:
+      retention: <time_specification> <1>
+      retentionSize: <size_specification> <2>
+----
++
+<1> Substitute `<time_specification>` with a number directly followed by `ms` (milliseconds), `s` (seconds), `m` (minutes), `h` (hours), `d` (days), `w` (weeks), or `y` (years).
+<2> Substitute `<size_specification>` with a number directly followed by `B` (bytes), `KB` (kilobytes), `KiB` (indicator kilobytes), `MB` (megabytes), `GB` (gigabytes), `TB` (terabytes), `PB` (petabytes), and `EB` (exabytes).
```

Comment:
MiB, GiB, TiB, PiB and EiB are all supported

Reply:
@bburt-rh  Once these comments addressed too changes look good to me

@simonpasquier want to take another look?

Reply:
remove `KiB` (indicator kilobytes), no need to mention MiB, GiB, TiB, PiB and EiB, 
`GB is the traditional, metric style of measurement with 1 GB equaling 1,000³ bytes. GiB is the binary method; which is the way computers measure data at 1024³ bytes.`
```
      --storage.tsdb.retention.size=STORAGE.TSDB.RETENTION.SIZE  
                                 Maximum number of bytes that can be stored for
                                 blocks. A unit is required, supported units: B,
                                 KB, MB, GB, TB, PB, EB. Ex: "512MB". Based on
                                 powers-of-2, so 1KB is 1024B. Use with server
                                 mode only.
```

Reply:
@slashpai According to above comments of @juzhao, the B and iB have the same meaning here, though our validator permits B and iB, it's ok for the document not mentioning iB to be consistent with upstream doc
https://github.com/openshift/cluster-monitoring-operator/blob/e1403615b67d069e2c32fcfed96d82291ff657a1/manifests/0000_50_cluster-monitoring-operator_00_0prometheus-custom-resource-definition.yaml#L5246

Reply:
Both types of units are supported by Prometheus our validator in Prometheus operator uses same validation as Prometheus. I am good with keeping docs similar to upstream